frontend/src/components/StockNews/StockNews.jsx

```diff
@@ -11,28 +11,49 @@ export default function StockNews({ stocks }) {
 
     useEffect(() => {
         const fetchStockNews = async () => {
-            if (currentStock) {
+            if (currentStock != null) {
                 try {
                     const searchTerm = firstWord(currentStock.name);
-                    const url = `https://newsapi.org/v2/everything?q=${searchTerm}&sortBy=relevancy&pageSize=5&apiKey=${import.meta.env.VITE_NEWS}`;
-                    const response = await axios.get(url);
-                    setStockNews(response.data.articles);
+                    const newsUrl = new URL("https://newsapi.org/v2/everything");
+                    newsUrl.searchParams.append("q", searchTerm);
+                    newsUrl.searchParams.append("sortBy", "relevancy");
+                    newsUrl.searchParams.append("pageSize", "5");
+                    newsUrl.searchParams.append("apiKey", import.meta.env.VITE_NEWS);
+                    const response = await axios.get(newsUrl);
+
+                    if (response.data.status === "ok") {
+                        setStockNews(response.data.articles);
+                    }
+
+                    if (response.data.status === "error") {
+                        setError(`{response.code}: {response.message}`);
+                    }
                 } catch (error) {
-                    setError(`${error.message}: Please try again later.`);
+                    setError(`System Error: ${error.message}`);
                 }
             }
         };
         fetchStockNews();
     }, [currentStock]);
 
+    const handleOptions = (event) => {
+        const ticker = event.target.value;
+        stocks.map((stock) => {
```

Comment:
looks like a bit of an antipattern --`.find` or `filter` may be more appropriate here since you're looking for the particular stock element that matches the given ticker